### ISSUE TITLE
Creating locations via geocoding

### ISSUE THREAD

**Issue by richardolsson** [zetkin/app.zetkin.org#1180]
## Description
A user should be able to create locations in the location modal using geocoding, which is the process by which a text string (usually an address) can be used to find a geographic location (coordinates).

## Relevant Job Stories
* When I need to create a new location, I want to be able to find it by searching for a nearby address, landmark etc, so that I don’t have to pan and zoom to the right location using a map

## Prerequisites
Don't start this until #1174 is done.

## Requirements
- [ ] When typing into the filter input, the app should search for matching locations globally
- [ ] All found items should be listed with a "Create location at " prefix
- [ ] If the user picks one of the "Create" options, go into create mode
- [ ] In the create mode, navigate to the location of the geocoded location
- [ ] In the create mode, prefill the title and description (street address) with the information from the geocoded location

## Possible implementations
Maybe use the Google geocoding API? As long as we use it only on the backend (via RPC), and don't include any user information, we should be fine. If there are any alternatives that are comparatively good, I would love to use them instead!

### Filtering
![image](https://user-images.githubusercontent.com/550212/228887268-54a078d1-d06e-4e1e-a253-1491dc9f7fef.png)

### Creating a new location
![image](https://user-images.githubusercontent.com/550212/228888043-6145f77a-cf75-41f5-9617-a75937e2af17.png)

## Open questions
None

**Comment by joe-irving** [comment]
Rather than using Google Maps you could use the Open Street Map API [Nominatim](https://nominatim.org/release-docs/latest/api/Search/) which could just be used directly on the front end. They [welcome external use](https://operations.osmfoundation.org/policies/nominatim/) so I imagine it would be fine. On every key up, just lookup something like this:

```
https://nominatim.openstreetmap.org/search?format=json&q=My%20House
```

Happy to play around with this

**Comment by richardolsson** [comment]
> Rather than using Google Maps you could use the Open Street Map API [Nominatim](https://nominatim.org/release-docs/latest/api/Search/) which could just be used directly on the front end. They [welcome external use](https://operations.osmfoundation.org/policies/nominatim/) so I imagine it would be fine. It does have less info than Google maps.

I appreciate the recommendation! I had tried some other alternatives, but in my experience, there just isn't anything that beats Google Maps API in this arena. I experimented briefly with nominatim and IMO it's not great at finding precise locations for addresses, but I'm very happy for you to play around and see if you can make it work.

**Comment by joe-irving** [comment]
Yeah you're right it isn't anywhere near as good as Google Maps. I am used to using it because it is free, and I guess there is a slight point of principle of using open source platforms. I messed around with using Nominatim on this and got the data coming through but there seems to be a bug in the Location Search tool with the autocomplete not rendering that i can't work out
